### PR TITLE
Feature/venue testcases

### PR DESCRIPTION
**Pull request by Backiaraj07**
Page load issue and update modification done in
Room,
Venue,
Ratecard,
Hotels,
Menus.
And added Taskdefinition file for AWS Auto deployment

### CODE REVIEW

File: BookingSystem.BAL/Interface/IRateCardService.cs [hunk 1]
```diff
@@ -9,9 +9,9 @@ namespace BookingSystem.BAL.Interface
 {
     public interface IRateCardService
     {
-        List<RateCardsVM> GetRateCardDetails();
+        Task<List<RateCardsVM>> GetRateCardDetails();
 
-        RateCardsVM GetRateCardDetailsByID(RateCardsVM _ratecardid);
+        Task<RateCardsVM> GetRateCardDetailsByID(int? _ratecardid);
```
Comment (sathish199308) [inline BookingSystem.BAL/Interface/IRateCardService.cs:14]: Hi Kindly not use _ in the prefix of the input parameters . Make it as a input parameters as RateCardID or rateCardId, kindly not use ratecardid

File: BookingSystem.BAL/Interface/IRateCardService.cs [hunk 2]
```diff
@@ -9,9 +9,9 @@ namespace BookingSystem.BAL.Interface
...
         bool UpdateRateCardDetails(RateCardsVM _ratecard);
 
         Task<RateCardsVM> SaveRateCardDetails(RateCardsVM _ratecard);
```
Comment (sathish199308) [inline BookingSystem.BAL/Interface/IRateCardService.cs:17]: Hi Kindly not use _ in the prefix of the input parameters

File: BookingSystem.BAL/Interface/IRoomService.cs [hunk 3]
```diff
@@ -11,7 +11,7 @@ public interface IRoomService
     {
         Task<List<RoomsVM>> GetRoomDetails();
 
-        RoomsVM GetRoomDetailsByID(RoomsVM _roomid);
+        Task<RoomsVM> GetRoomDetailsByID(int? _roomid);
```
Comment (sathish199308) [inline BookingSystem.BAL/Interface/IRoomService.cs:14]: Hi Kindly not use _ in the prefix of the input parameters . Make it as a input parameters as RoomID or roomId, kindly not use roomid

File: BookingSystem.BAL/Interface/IRoomService.cs [hunk 4]
```diff
@@ -11,7 +11,7 @@ public interface IRoomService
...
         bool UpdateRoomDetails(RoomsVM _room);
 
         Task<RoomsVM> SaveRoomDetails(RoomsVM _room);
```
Comment (sathish199308) [inline BookingSystem.BAL/Interface/IRoomService.cs:17]: Hi Kindly not use _ in the prefix of the input parameters .

File: BookingSystem.BAL/Interface/IRateCardService.cs (hunk 1 above)
Comment (vivekr2021) [reply BookingSystem.BAL/Interface/IRateCardService.cs:14]: Please practice adding Async as the suffix for Async methods
Instead "GetRateCardDetailsByID" use "GetRateCardDetailsByIDAsync"

File: BookingSystem.BAL/Interface/IHotelService.cs [hunk 5]
```diff
@@ -9,9 +9,9 @@ namespace BookingSystem.BAL.Interface
 {
     public interface IHotelService
     {
-        List<HotelVM> GetHotelDetails();
+        Task<List<HotelVM>> GetHotelDetails();
 
-        HotelVM GetHotelDetailsByID(HotelVM HotelID);
+        Task<HotelVM> GetHotelDetailsByID(int? HotelID);
```
Comment (vivekr2021) [inline BookingSystem.BAL/Interface/IHotelService.cs:14]: Use Async in the suffix for Async methods
Use "GetHotelDetailsByIDAsync" Instead "GetHotelDetailsByID"

**Review by vivekr2021 — CHANGES_REQUESTED** [review]

File: BookingSystem.BAL/Service/VenueService.cs [hunk 6]
```diff
@@ -21,28 +21,28 @@ public bool DeleteVenueDetails(VenuesVM _venueId)
...
-            throw new NotImplementedException();
+            return await _venueRepository.GetVenueDetailsByID(_venueId).ConfigureAwait(false);
         }
 
         public async Task<VenuesVM> SaveVenueDetails(VenuesVM _venue)
         {
-            if(_venue.VenuesID>0)
+            if (_venue.VenuesID > 0)
             {
-               await _venueRepository.UpdateVenue(_venue);
```
Comment (manic1990) [inline BookingSystem.BAL/Service/VenueService.cs:38]: Use ConfigureAwait(false) both method

File: BookingSystem.BAL/Service/VenueService.cs [hunk 7]
```diff
@@ -21,28 +21,28 @@ public bool DeleteVenueDetails(VenuesVM _venueId)
...
         {
-            if(_venue.VenuesID>0)
+            if (_venue.VenuesID > 0)
             {
-               await _venueRepository.UpdateVenue(_venue);                
+                await _venueRepository.UpdateVenue(_venue);
             }
             else
             {
                 var venues = await _venueRepository.saveVenue(_venue);
```
Comment (manic1990) [inline BookingSystem.BAL/Service/VenueService.cs:42]: Use ConfigureAwait(false).

File: BookingSystem.BAL/Interface/IHotelService.cs [hunk 8]
```diff
@@ -9,9 +9,9 @@ namespace BookingSystem.BAL.Interface
...
         bool UpdateHotelDetails(HotelVM _Hotel);
```
Comment (manic1990) [inline BookingSystem.BAL/Interface/IHotelService.cs:15]: Use Task<bool >

File: BookingSystem.BAL/Interface/IRateCardService.cs [hunk 9]
```diff
@@ -9,9 +9,9 @@ namespace BookingSystem.BAL.Interface
 {
     public interface IRateCardService
     {
-        List<RateCardsVM> GetRateCardDetails();
+        Task<List<RateCardsVM>> GetRateCardDetails();
 
-        RateCardsVM GetRateCardDetailsByID(RateCardsVM _ratecardid);
+        Task<RateCardsVM> GetRateCardDetailsByID(int? _ratecardid);
         bool UpdateRateCardDetails(RateCardsVM _ratecard);
```
Comment (manic1990) [inline BookingSystem.BAL/Interface/IRateCardService.cs:15]: Use Task for async method

File: BookingSystem.BAL/Interface/IRoomService.cs [hunk 10]
```diff
@@ -11,7 +11,7 @@ public interface IRoomService
     {
         Task<List<RoomsVM>> GetRoomDetails();
 
-        RoomsVM GetRoomDetailsByID(RoomsVM _roomid);
+        Task<RoomsVM> GetRoomDetailsByID(int? _roomid);
         bool UpdateRoomDetails(RoomsVM _room);
```
Comment (manic1990) [inline BookingSystem.BAL/Interface/IRoomService.cs:15]: Use Task for async method

File: BookingSystem.BAL/Interface/IVenueService.cs [hunk 11]
```diff
@@ -9,9 +9,9 @@ namespace BookingSystem.BAL.Interface
 {
    public interface IVenueService
     {
-        List<VenuesVM> GetVenueDetails();
+        Task<List<VenuesVM>> GetVenueDetails();
 
-        VenuesVM GetVenueDetailsByID(int _venueId);
+        Task<VenuesVM> GetVenueDetailsByID(int _venueId);
         bool UpdateVenueDetails(VenuesVM _venue);
```
Comment (manic1990) [inline BookingSystem.BAL/Interface/IVenueService.cs:15]: Use Task for async method

File: BookingSystem.BAL/Interface/IRateCardService.cs (hunk 1 above)
Comment (Backiaraj07) [reply BookingSystem.BAL/Interface/IRateCardService.cs]: Sure sir. Changes done.

File: BookingSystem.BAL/Interface/IRateCardService.cs (hunk 2 above)
Comment (Backiaraj07) [reply BookingSystem.BAL/Interface/IRateCardService.cs]: Changes done

File: BookingSystem.BAL/Interface/IRoomService.cs (hunk 4 above)
Comment (Backiaraj07) [reply BookingSystem.BAL/Interface/IRoomService.cs]: Changes done

File: BookingSystem.BAL/Interface/IRoomService.cs (hunk 3 above)
Comment (Backiaraj07) [reply BookingSystem.BAL/Interface/IRoomService.cs]: Changes done

File: BookingSystem.BAL/Interface/IHotelService.cs (hunk 5 above)
Comment (Backiaraj07) [reply BookingSystem.BAL/Interface/IHotelService.cs]: Sure sir.

File: BookingSystem.BAL/Service/VenueService.cs (hunk 6 above)
Comment (Backiaraj07) [reply BookingSystem.BAL/Service/VenueService.cs:38]: Changes done

File: BookingSystem.BAL/Interface/IHotelService.cs (hunk 8 above)
Comment (Backiaraj07) [reply BookingSystem.BAL/Interface/IHotelService.cs]: Changes done

File: BookingSystem.BAL/Interface/IRateCardService.cs (hunk 9 above)
Comment (Backiaraj07) [reply BookingSystem.BAL/Interface/IRateCardService.cs]: Changes done

File: BookingSystem.BAL/Interface/IRoomService.cs (hunk 10 above)
Comment (Backiaraj07) [reply BookingSystem.BAL/Interface/IRoomService.cs]: Changes done

File: BookingSystem.BAL/Interface/IVenueService.cs (hunk 11 above)
Comment (Backiaraj07) [reply BookingSystem.BAL/Interface/IVenueService.cs]: Changes done

**Review by sathish199308 — APPROVED** [review]
Approved the PR

**Pull request merged**